File: modules/agent/src/schemas/conversation.py
"""Conversation schemas for API validation."""

from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class ConversationCreate(BaseModel):
    thread_id: UUID
    title: Optional[str] = None


class ConversationUpdate(BaseModel):
    title: str


class MessageCreate(BaseModel):
    role: str
    content: str
    token_usage: Optional[dict] = None


class MessageResponse(BaseModel):
    id: int
    role: str
    content: str
    token_usage: Optional[dict]
    created_at: datetime

    class Config:
        from_attributes = True


class UserBrief(BaseModel):
    id: int
    email: Optional[str] = None

    class Config:
        from_attributes = True


class ConversationResponse(BaseModel):
    id: int
    thread_id: UUID
    title: Optional[str]
    created_at: datetime
    updated_at: datetime
    archived_at: Optional[datetime]

    class Config:
        from_attributes = True


class ConversationWithUserResponse(ConversationResponse):
    """Conversation response with user details for full listing."""
    created_by_id: Optional[int] = None
    updated_by_id: Optional[int] = None
    created_by: Optional[UserBrief] = None
    updated_by: Optional[UserBrief] = None


class ConversationWithMessagesResponse(BaseModel):
    conversation: ConversationResponse
    messages: list[MessageResponse]

File: modules/agent-go/internal/schemas/conversation.go
package schemas

import (
	"time"

	"github.com/google/uuid"
)

// ConversationCreate represents the request body for creating a conversation
type ConversationCreate struct {
	ThreadID uuid.UUID `json:"thread_id" validate:"required"`
	Title    *string   `json:"title"`
}

// ConversationUpdate represents the request body for updating a conversation
type ConversationUpdate struct {
	Title string `json:"title" validate:"required"`
}

// MessageCreate represents the request body for creating a message
type MessageCreate struct {
	Role       string         `json:"role" validate:"required"`
	Content    string         `json:"content" validate:"required"`
	TokenUsage map[string]any `json:"token_usage"`
}

// MessageResponse represents a message in API responses
type MessageResponse struct {
	ID         int64          `json:"id"`
	Role       string         `json:"role"`
	Content    string         `json:"content"`
	TokenUsage map[string]any `json:"token_usage"`
	CreatedAt  time.Time      `json:"created_at"`
}

// UserBrief represents a brief user summary in API responses
type UserBrief struct {
	ID    int64   `json:"id"`
	Email *string `json:"email"`
}

// ConversationResponse represents a conversation in API responses
type ConversationResponse struct {
	ID         int64      `json:"id"`
	ThreadID   uuid.UUID  `json:"thread_id"`
	Title      *string    `json:"title"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`
	ArchivedAt *time.Time `json:"archived_at"`
}

// ConversationWithUserResponse represents a conversation with user details
type ConversationWithUserResponse struct {
	ConversationResponse
	CreatedByID *int64     `json:"created_by_id"`
	UpdatedByID *int64     `json:"updated_by_id"`
	CreatedBy   *UserBrief `json:"created_by"`
	UpdatedBy   *UserBrief `json:"updated_by"`
}

// ConversationWithMessagesResponse represents a conversation with its messages
type ConversationWithMessagesResponse struct {
	Conversation ConversationResponse `json:"conversation"`
	Messages     []MessageResponse    `json:"messages"`
}
